feat: add sdks and libraries section

jboix opened the pull request:
## Description

Resolves #18 by introducing the SDK and Libraries section to provide a clear framework for integrating SDKs and libraries across various platforms. The changes include:

## Changes Made

1. An **Introduction** article that outlines the purpose of this section and the importance of maintaining a coherent ecosystem of development tools. It includes a decision flowchart to visually guide the integration process of new SDKs or libraries.

2. **Platform-specific template** for documenting recommended SDKs and libraries. Each template features a prioritization matrix to categorize the tools based on their necessity and utility.
## Checklist:

- [x] I have followed the style guidelines of this project.
- [x] I have performed a self-review of my own changes.
- [x] I have made corresponding changes to the documentation.
- [x] I have reviewed the [CONTRIBUTING.md](../docs/CONTRIBUTING.md) guidelines.

jboix committed:
feat: add sdks and libraries section

Resolves #18 by introducing the SDK and Libraries section to provide a clear framework for
integrating SDKs and libraries across various platforms. The changes include:

1. An **Introduction** article that outlines the purpose of this section and the importance of
   maintaining a coherent ecosystem of development tools. It includes a decision flowchart to
   visually guide the integration process of new SDKs or libraries.

2. **Platform-specific template** for documenting recommended SDKs and libraries. Each template
   features a prioritization matrix to categorize the tools based on their necessity and utility.

defagos commented on public/_sidebar.md:
```diff
@@ -10,3 +10,7 @@
 * [Code of Conduct Guidelines](/guides/CODE_OF_CONDUCT_GUIDE.md)
 * [Security policy Guidelines](/guides/SECURITY_GUIDE.md)
+
+**SDK and Libraries**
```
```suggestion
**SDKs and Libraries**
```

defagos commented on public/sdks/INTRODUCTION.md:
```diff
@@ -0,0 +1,74 @@
+# SDK and Libraries
```
```suggestion
# SDKs and Libraries
```

MGaetan89 commented on public/_sidebar.md:
```diff
@@ -10,3 +10,7 @@
+
+**SDK and Libraries**
+
+* [Introduction](/sdks/INTRODUCTION.md)
```
Will/should the platform specific pages be listed here?

jboix replied:
Yes, but since it's just a template for the moment I haven't put anything on the sidebar. I'm also thinking about removing the links from the `Introduction` page before merging. It makes sense to add them only once they are ready.

jboix replied:
I've added them as a comment and commented out the `platform specific` section in the `Introduction` until at least an article is available.

jboix committed:
fix: typo in the article title

Changed `SDK` to `SDKs`.

Co-authored-by: Samuel Défago <defagos@gmail.com>

jboix committed:
fix: comment out unavaible sections

Removes the platform specific section from the SDKs and Libraries Introduction
and comments out the access in the sidebar. The sections will be restored once
at least an article is available.
